cirq/contrib/quirk/quirk_gate.py

```diff
@@ -54,9 +57,35 @@ def same_half_turns(a1: float, a2: float, atol=0.0001) -> bool:
     return abs(d) < atol
 
 
-def angle_to_exponent_key(t: Union[float, sympy.Symbol]) -> Optional[str]:
+def _is_supported_formula(formula: sympy.Basic):
+    if isinstance(formula, (sympy.Symbol, sympy.Integer, sympy.Float,
+                            sympy.Rational, sympy.NumberSymbol)):
+        return True
+    if isinstance(formula, (sympy.Add, sympy.Mul)):
+        return all(_is_supported_formula(f) for f in formula.args)
+    return False
+
+
+def _val_to_quirk_formula(t: Union[float, sympy.Basic]) -> Optional[str]:
+    if isinstance(t, sympy.Basic):
+        if not set(t.free_symbols) <= {sympy.Symbol('t')}:
```

Comment:
Optional: This conditional is part of our validation logic, so it fits better in `_is_supported_formula`. I understand you've placed it here to generate more informative exceptions. An alternative that gives you both is a function `_check_formula_supported` that doesn't return anything but raises an exception on unsupported formulas. Up to you.

Reply:
Acknowledged.